rmc/utils/generic.py

```diff
@@ -497,77 +509,194 @@ def get_constraint_transcripts(outlier: bool = True) -> hl.expr.SetExpression:
+
+    Function imports variants from TSV into HT, removes malformed rows,
+    and lifts data from GRCh38 to GRCh37.
+
+    :param overwrite: Whether to overwrite Table if it exists.
+    :return: None; Function writes Table to temporary path.
+    """
+    fu_ht = hl.import_table(
+        asc_ssc_spark_de_novo_tsv,
+        impute=True,
+        # Skip blank lines at the bottom of this TSV
+        missing="",
+        skip_blank_lines=True,
+    )
+    # Remove lines from bottom of TSV that are parsed incorrectly upon import
+    # These lines contain metadata about the TSV, e.g.:
+    # "Supplementary Table 20. The de novo SNV/indel variants used in TADA
+    # association analyses from assembled ASD cohorts"
+    fu_ht = fu_ht.filter(hl.is_missing(fu_ht.Role))
+
+    # Prepare to lift dataset back to GRCh37
+    rg37 = hl.get_reference("GRCh37")
+    rg38 = hl.get_reference("GRCh38")
+    rg38.add_liftover(
+        "gs://hail-common/references/grch38_to_grch37.over.chain.gz", rg37
+    )
+
+    fu_ht = fu_ht.annotate(
+        new_locus=hl.liftover(fu_ht.locus, "GRCh37", include_strand=True),
+        old_locus=fu_ht.locus,
+    )
+    liftover_stats = fu_ht.aggregate(
+        hl.struct(
+            liftover_fail=hl.agg.count_where(hl.is_missing(fu_ht.new_locus)),
+            flip_strand=hl.agg.count_where(fu_ht.new_locus.is_negative_strand),
+        )
+    )
+    logger.info(
+        "%i variants failed to liftover, and %i variants flipped strands",
+        liftover_stats.liftover_fail,
+        liftover_stats.flip_strand,
+    )
+    fu_ht = fu_ht.filter(
+        hl.is_defined(fu_ht.new_locus) & ~fu_ht.new_locus.is_negative_strand
+    )
+    fu_ht = fu_ht.key_by(locus=fu_ht.new_locus.result, alleles=fu_ht.alleles)
+    fu_ht = fu_ht.group_by("locus", "alleles").aggregate(
+        role=hl.agg.collect_as_set(fu_ht.Role),
+        samples=hl.agg.collect_as_set(fu_ht.Sample),
+    )
+    fu_ht.write(f"{TEMP_PATH_WITH_FAST_DEL}/fu_dn.ht", overwrite=overwrite)
+
+
+def import_kaplanis_data(overwrite: bool) -> None:
+    """
+    Import de novo variants from Kaplanis et al. (2020) paper.
+
+    Function imports variants from TSV into HT, and filters to cases with
+    neurodevelopmental disorders only.
+    Input TSV also contains autistic individuals, but these individuals overlap with
+    data from Fu et al. paper.
+
+    :param overwrite: Whether to overwrite Table if it exists.
+    :return: None; Function writes Table to temporary path.
+    """
+    kap_ht = hl.import_table(ddd_autism_de_novo_tsv, impute=True)
+    kap_ht = kap_ht.transmute(
+        locus=hl.locus(kap_ht.chrom, kap_ht.pos),
+        alleles=[kap_ht.ref, kap_ht.alt],
+    )
+    kap_ht = kap_ht.filter(hl.is_snp(kap_ht.alleles[0], kap_ht.alleles[1]))
+    kap_ht = kap_ht.group_by("locus", "alleles").aggregate(
+        study=hl.agg.collect(kap_ht.study),
+        case_control=hl.agg.collect(kap_ht.case_control),
+    )
```

Comment:
I don't see where a filter was applied here to just individuals with DD/ID?

Reply:
I probably forgot to copy it from the notebook, will add